frame/nomination-pools/src/lib.rs

```diff
@@ -686,6 +794,82 @@ impl<T: Config> BondedPool<T> {
 			.saturating_sub(T::StakingInterface::active_stake(&account).unwrap_or_default())
 	}
 
+	/// Set the pool's commission.
+	fn set_commission_current(mut self, commission: &Perbill, payee: T::AccountId) -> Self {
+		// Force a commission of `None` if a 0% commission is provided.
+		// Note that a 0% commission will replace the whole tuple with None.
+		let new_current =
+			if commission > &Perbill::from_percent(0) { Some((*commission, payee)) } else { None };
+
+		// update commission if `Some(commission)` is to be inserted, or `None` otherwise.
+		self.commission = match &self.commission {
+			Some(c) => Some(Commission { current: new_current, ..c.clone() }),
+			None => match new_current.is_some() {
+				true => Some(Commission { current: new_current, ..Commission::default() }),
+				false => None,
+			},
+		};
+
+		// if throttle is present, record the current block as the previously
+		// updated commission.
+		if let Some(c) = &self.commission {
+			if let Some(throttle) = &c.throttle {
+				self.commission = Some(Commission {
+					throttle: Some(CommissionThrottle {
+						previous_set_at: Some(<frame_system::Pallet<T>>::block_number()),
+						..*throttle
+					}),
+					..c.clone()
+				});
+			}
+		}
+		self
+	}
+
+	/// Set the pool's maximum commission.
+	fn set_max_commission(mut self, max_commission: Perbill) -> Self {
```

Comment:
1. not sure why this function has to consume the whole `BondedPool` and re-return it again? `&mut self` is the way to go.

Reply:
2. and I think the first step in making it better is defining what the behavior should be: what should this function do if commission does not exist? silently do nothing? or return an error? 

answering this will probably help you rewrite it better.